text2chem/core/formula_parser.py

```diff
@@ -92,7 +92,7 @@ def __parse_parentheses(init_formula, init_factor, curr_dict):
         else:
             curr_dict[el] = amt
 
-    return curr_dict
+    return curr_dict, bool(unit_sym_dict)
```

Comment:
Sorry, could you please explain a bit how this change works? Why do you need unit_sym_dict explicitly as boolean?

Reply:
This boolean value is used to control the replacement operation in the **outer layer** when the **inner layer** nested parentheses regex matches multiplication factors, the replacement operation will be different.

Consider this example: '(Cu1/3Nb2/3)1/4-yTi3/4-xZr(x+y)O2'

When reaching the second subtree and matching '(x+y)', after the recursion fails to match, the function `__get_sym_dict` interprets and does not update `unit_sym_dict={}`. `bool(unit_sym_dict)=False`, so the outer layer only needs to replace the expression with **'Ti3/4-xZrx+yO2'**,  otherwise with **'Ti3/4-xZrO2'**. The latter is wrong.

May I ask if my explanation is clear?

Reply:
Thanks, it's clearer now.